src/App.tsx

```diff
@@ -14,11 +15,114 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  isListVisible: boolean;
+  listOfGoods: string[];
+};
+
+class App extends React.PureComponent<{}, State> {
+  state: State = {
+    isListVisible: false,
+    listOfGoods: goodsFromServer,
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      listOfGoods: [...state.listOfGoods].reverse(),
+    }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState(state => ({
+      listOfGoods: [...state.listOfGoods].sort((a, b) => a.localeCompare(b)),
+    }));
+  };
+
+  sortbyLength = () => {
+    this.setState(state => ({
+      listOfGoods: [...state.listOfGoods].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      listOfGoods: [...goodsFromServer],
+    });
+  };
+
+  render() {
+    const { isListVisible, listOfGoods } = this.state;
+
+    return (
+      <div className="App">
+        <ul
+          className={
+            classNames(
+              'list',
+              { 'list-visible': isListVisible === true },
```

Comment:
You can write this expression more concisely. Because "isVisible" is boolean and css class name "list-visible" can be written without quotes